src/main/resources/assets/computercraft/lua/rom/apis/rednet.lua

```diff
@@ -93,10 +99,19 @@ function send( nRecipient, message, sProtocol )
 end
 
 function broadcast( message, sProtocol )
+    if sProtocol ~= nil and type( sProtocol ) ~= "string" then
+        error( "bad argument #2 (expected string, got " .. type( sProtocol ) .. ")", 2 )
+    end
     send( CHANNEL_BROADCAST, message, sProtocol )
 end
 
 function receive( sProtocolFilter, nTimeout )
+    if sProtocolFilter ~= nil and type( sProtocolFilter ) ~= "string" then
+        error( "bad argument #1 (expected string, got " .. type( sProtocolFilter ) .. ")", 2 )
+    end
+    if nTimeout ~= nil and type( nTimeoutl ) ~= "number" then
+        error( "bad argument #2 (expected number, got " .. type( nTimeout ) .. ")", 2 )
+    end
     -- The parameters used to be ( nTimeout ), detect this case for backwards compatibility
     if type(sProtocolFilter) == "number" and nTimeout == nil then
```

Comment:
The check above makes this line pointless. It would make more sense to move checks after this if to ensure backwards compatibility.